avail-server/src/main/kotlin/com/avail/server/io/AbstractTransportChannel.kt

```diff
@@ -49,24 +49,25 @@ import java.util.*
  * @author Richard Arriaga &lt;rich@availlang.org&gt;
  * @param T
  *   The type of the enclosed channel.
+ *
+ * @constructor
+ * Construct an [AbstractTransportChannel].
+ *
+ * @param closeAction
+ *   The custom action that is to be called when the this channel is closed in
+ *   order to support implementation-specific requirements after the closing of
+ *   this channel.
  */
-abstract class AbstractTransportChannel<T> : AvailServerChannel()
+abstract class AbstractTransportChannel<T> constructor(
+		closeAction: (DisconnectReason, AvailServerChannel) -> Unit)
```

Comment:
This should probably default to `{}`. Even if the `AvailServer` subsystem always supplies an actual callback, the API is intended for broader use, and "do nothing" will occasionally be a valid use case.